test/ApplicationTest.php

```diff
@@ -265,6 +265,35 @@ public function testFinalHandlerIsUsedAtInvocationIfNoOutArgumentIsSupplied()
         $this->assertSame($finalResponse, $test);
     }
 
+    public function testFinalHandlerCreatedAtInvocationIsProvidedResponseInstance()
+    {
+        $routeResult = RouteResult::fromRouteFailure();
+        $this->router->match()->willReturn($routeResult);
+
+        $app = new Application($this->router->reveal());
+
+        $finalResponse = $this->prophesize('Psr\Http\Message\ResponseInterface');
```

Comment:
What about using ::class in your tests too ? :) At least you will get refactoring everywhere